jobserver/api.py

```diff
@@ -137,13 +138,50 @@ def post(self, request, *args, **kwargs):
                 # related Jobs manager (ie job_request.jobs)
                 job_data.pop("job_request_id")
 
-                job, created = job_request.jobs.update_or_create(
+                job, created = job_request.jobs.get_or_create(
                     identifier=job_data["identifier"],
                     defaults={**job_data},
                 )
-
                 log.info("Created or updated Job", job=job.id, created=created)
 
+                if created:
+                    # this is the first time job-server has heard about this
+                    # Job so we can move on to further Jobs.  We're knowingly
+                    # skipping potential notifications here to avoid creating
+                    # false positives.
+                    continue
+
+                # check to see if the Job is about to transition to finished
+                # (failed or succeeded) so we can notify after the update
+                finished = ["failed", "succeeded"]
+                should_notify = (
+                    job.status not in finished and job_data["status"] in finished
+                )
```

Comment:
I wonder if an alternative approach would be to add a `notified` flag to Job itself. Given that the existing view logic just updates the fields from job-runner and leaves existing fields alone I don't think this would need any special handling. That would handle the case of jobs which start in the failed state.

Reply:
As in track that we've sent a notification for the given Job?

My one concern, and it's a pretty weak one, would be changing the scope of job-runner's ownership.  Currently it's "all of Job" (let's just handwave away the PK for now…), so adding fields to it which job-runner didn't control could add complexity/loaded footguns for later.

Reply:
Second thought!  We'll still need to make this check since we need to know when to notify, separate from "have we notified about this before?".